pkgs/development/python-modules/hatchling/default.nix

```diff
@@ -0,0 +1,45 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+
+# All runtime dependencies
+, editables
+, importlib-metadata
+, packaging
+, pathspec
+, pluggy
+, tomli
+}:
+
+buildPythonPackage rec {
+  pname = "hatchling";
+  version = "0.19.0";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "942b34e454ce3be5b03139e0a198ba8ebc08a6095d345fe829b1ae92e9326c2c";
+  };
+
+  propagatedBuildInputs = [
+    editables
+    packaging
+    pathspec
+    pluggy
+    tomli
```

Comment:
`nix-build . -A python3Packages.hatchling`

build fails with
```
ERROR: Could not find a version that satisfies the requirement tomli~=2.0.0; python_version > "3" (from hatchling) (from versions: none)
ERROR: No matching distribution found for tomli~=2.0.0; python_version > "3"
```

sadly the tomli update pr hasn't been merged and it also will go in to staging which will take some time to reach the master branch https://github.com/NixOS/nixpkgs/pull/157735

i will test the builds and if they succeed then merge


so we can either leave this unmerged and then i merge this once the tomli pr is in master or i merge this in to staging

BTW afaik the hatchling does not need to be in nixpkgs for poetry2nix to be able to use it though since hatchling is not poetry, poetry2nix will need to add support for it